### ISSUE TITLE
Source Sendgrid: new stream Unsubscribe Groups

### ISSUE THREAD

**Issue by igrankova** [airbytehq/airbyte#17460]
## Tell us about the problem you're trying to solve
[Unsubscribe Groups](https://docs.sendgrid.com/api-reference/suppressions-unsubscribe-groups/retrieve-all-suppression-groups-associated-with-the-user)
Suppression groups, or unsubscribe groups, are specific types or categories of emails from which you would like your recipients to be able to unsubscribe. For example: Daily Newsletters, Invoices, and System Alerts are all potential suppression groups. Visit the main documentation to [learn more about suppression/unsubscribe groups](https://sendgrid.com/docs/ui/sending-email/unsubscribe-groups/).

[API key permissions](https://docs.sendgrid.com/ui/account-and-settings/api-keys#api-key-permissions)
[Types of API keys](https://docs.sendgrid.com/ui/account-and-settings/api-keys#types-of-api-keys)

## Describe the solution you’d like
Research pros and cons for implementing this stream.
_Research should contain: what stream should be implemented, permissions and scopes (if needed), paid plan (if needed)._

**Comment by btkcodedev** [comment]
@YowanR I would like to take this issue

**Comment by YowanR** [comment]
@btkcodedev are you still interested in this one, please?

**Comment by btkcodedev** [comment]
Nope, I've got other issues to take care of, it's free to claim now

**Comment by haithem-souala** [comment]
I will take this one!

**Comment by justinbchau** [comment]
Just assigned to you! Thanks for taking this @haithem-souala

**Comment by haithem-souala** [comment]
Hey @YowanR, the PR is ready to review.
